### ISSUE TITLE
Flashing firmware with .wav file

### ISSUE THREAD

**Issue by TotalRefusal** [jakplugg/Typhoon#10]
Does anybody know if flashing firmware with .wav files is an option for Typhoon??? Don´t make it to work with blinking Freeze on startup....

**Comment by chmanie** [comment]
Yes, I did that successfully with the microcell and superparasites firmware. Press the freeze button while switching on power, then play the .wav file into an input. Make sure no other cables are patched. The freeze LED should blink frantically and after the wave is gone through the module restarts.

**Comment by astrzepka** [comment]
It doesn't work for me either. I power up my system with Freeze button pressed, but module seems to power up as usual. Freeze blinks, but it also enters this mode if I hold it with module already powered up.
Playing .wav gives no result.

**Comment by n0kulture** [comment]
Same problem here. I've tried to update/upgrade firmware but it doesn't work. The module enters in upgrade mode, but when I play the .wav file it didn't make any change, and when the file end, the module doesn't reboot. In original clouds when I send the .wav file the freeze button start to flash in different frequency. 
So someone knows how to solve this problem?

**Comment by dragosandriana** [comment]
> Same problem here. I've tried to update/upgrade firmware but it doesn't work. The module enters in upgrade mode, but when I play the .wav file it didn't make any change, and when the file end, the module doesn't reboot. In original clouds when I send the .wav file the freeze button start to flash in different frequency.
> So someone knows how to solve this problem?

Make sure you followed the procedure and that the wav file has uploaded until the end. Is very important that the module receives the whole wav file and doesn't stop responding while the file is playing. If it doesn't work, try with another player, cell phone worked for me. And different volumes.

**Comment by n0kulture** [comment]
> Make sure you followed the procedure and that the wav file has uploaded until the end. Is very important that the module receives the whole wav file and doesn't stop responding while the file is playing. If it doesn't work, try with another player, cell phone worked for me. And different volumes.

No way, I've tried with my phone, my computer and differents players and volumes... but the module's behavior is like receive nothing even the VU leds were lighted... but the freeze LED flash all the time to same frequency. And yes, when the file end, the module continue to flash freeze led and no reboot.

**Comment by dragosandriana** [comment]
> > Make sure you followed the procedure and that the wav file has uploaded until the end. Is very important that the module receives the whole wav file and doesn't stop responding while the file is playing. If it doesn't work, try with another player, cell phone worked for me. And different volumes.
> 
> No way, I've tried with my phone, my computer and differents players and volumes... but the module's behavior is like receive nothing even the VU leds were lighted... but the freeze LED flash all the time to same frequency. And yes, when the file end, the module continue to flash freeze led and no reboot.

Try to write the hex clouds file with STLink to see if anything's wrong with the module.

**Comment by astrzepka** [comment]
I tried every button I found on the module, it just doesn't go into bootloader mode so I can play .wav file to it.
I don't have JTAG programmer, found out that there's loaded wrong firmware to module which makes all CV inverted. I want to flash it with correct one but it just doesn't want to get into loader mode. UI behaves as I wasn't holding FREEZE button, after a sec it starts blinking indicating that sample playback is in reverse mode. Playing .wav file does nothing at any point of startup, button combo, etc.

Is there anything I can do without access to JTAG flasher?

**Comment by dragosandriana** [comment]
> I tried every button I found on the module, it just doesn't go into bootloader mode so I can play .wav file to it.
> I don't have JTAG programmer, found out that there's loaded wrong firmware to module which makes all CV inverted. I want to flash it with correct one but it just doesn't want to get into loader mode. UI behaves as I wasn't holding FREEZE button, after a sec it starts blinking indicating that sample playback is in reverse mode. Playing .wav file does nothing at any point of startup, button combo, etc.
> 
> Is there anything I can do without access to JTAG flasher?

You can't upload (play) a wav file on an empty or wrongly written firmware. First you have to upload the official hex clouds fw.
Then power the module with "freeze" button pressed. After this, the "freeze" LED should blink. If it doesn't blink, re-write the original hex fw using jtag pins, with ST-Link v2 (or some similar device). There is no way around this. On the other hand, if you're in DIY and want to make some Mutable Instruments modules without having ST-Link programmer, you'd better forget about it. Is a cheap and essential device.
